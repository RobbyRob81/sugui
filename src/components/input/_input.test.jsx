import * as React from 'react';
import { customRender, fireEvent } from '../../../test/test-utils';
import Input from '.';

it('should render without throwing an error', () => {
  const { container } = customRender(<Input />);
  expect(container.firstChild).toMatchSnapshot();
});

it('Should allow the user to enter data', async () => {
  const { getByPlaceholderText } = customRender(<Input inputProps={{ placeholder: 'test' }} />);
  const inputNode = getByPlaceholderText('test');

  fireEvent.change(inputNode, { target: { value: 'lion' } });
  expect(inputNode.value).toBe('lion');
});

it('calls "onChange" prop when the input value changes', () => {
  const onChange = jest.fn();
  const { getByPlaceholderText } = customRender(<Input onChange={onChange} inputProps={{ placeholder: 'test' }} />);
  const inputNode = getByPlaceholderText('test');

  fireEvent.change(inputNode, { target: { value: 'lion' } });
  expect(onChange).toHaveBeenCalled();
});
