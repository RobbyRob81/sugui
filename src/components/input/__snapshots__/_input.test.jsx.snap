// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  class="sc-bdVaJa erLRLT"
>
  <input
    class="sc-bwzfXH LXLUC"
    color="porcelain"
  />
</div>
`;
